src/machine/machine_avr.go

```diff
@@ -58,3 +58,93 @@ func (p GPIO) Get() bool {
 		return (val > 0)
 	}
 }
+
+const (
+	COM2A1 = 7
+	COM2A0 = 6
+	COM2B1 = 5
+	COM2B0 = 4
+	COM0A1 = 7
+	COM0A0 = 6
+	COM0B1 = 5
+	COM0B0 = 4
+	COM1A1 = 7
+	COM1A0 = 6
+	COM1B1 = 5
+	COM1B0 = 4
+
+	CS00 = 0
+	CS01 = 1
+	CS11 = 1
+	CS22 = 2
+
+	WGM10 = 0
+	WGM11 = 1
+	WGM20 = 0
+	WGM21 = 1
+)
+
+// InitPWM initializes the registers needed for PWM.
+func InitPWM() {
+	// use waveform generation
+	*avr.TCCR0A |= avr.TCCR0A_WGM0
+
+	// set timer 0 prescale factor to 64
+	*avr.TCCR0B |= 1 << CS01
+	*avr.TCCR0B |= 1 << CS00
+
+	// set timer 1 prescale factor to 64
+	*avr.TCCR1B |= 1 << CS11
+
+	// put timer 1 in 8-bit phase correct pwm mode
+	*avr.TCCR1A |= 1 << WGM10
+
+	// set timer 2 prescale factor to 64
+	*avr.TCCR2B |= 1 << CS22
+
+	// configure timer 2 for phase correct pwm (8-bit)
+	*avr.TCCR2A |= 1 << WGM20
+}
+
+// Configure configures a PWM pin for output.
+func (pwm PWM) Configure() {
+	if pwm.Pin < 8 {
+		*avr.DDRD |= 1 << pwm.Pin
+	} else {
+		*avr.DDRB |= 1 << (pwm.Pin - 8)
+	}
+}
+
+// Set sets the needed register values to turn on the duty cycle for a PWM pin.
+func (pwm PWM) Set(value uint8) {
+	switch pwm.Pin {
+	case 3:
```

Comment:
These numbers should be constants - something like:
```go
const (
    PWM0 = 3
    PWM1 = 5
    // etc
)
```
Constants are also easier to share with other chips, e.g. the nrf52 also has PWM but on different pins.

Reply:
The basic philosophy I have tried try to follow is to use whatever is silkscreened on the circuit board. 

Since the Arduino labels the board itself with the pin numbers  "3", "5", etc. I figured it would be a lot easier for users to determine how to write the code that does what they expect. Creating a separate naming/indexing system aka PWM0 etc will require another lookup of the docs for a dev to figure out that PWM0 means pin 3 on Arduino, and some other pin on nrf52.

Perhaps it would be better to have separate example for PWM on nrf52 and eventually any other platforms that are added that support a PWM interface.

Reply:
Ah, I mixed ADC and PWM - ADC is numbered differently. Yes, then it makes sense to keep these numbers. It's a bit unfortunate that it is hard to share example code across platforms but so be it.

I also don't really like how these numbers are hardcoded for all AVR chips (in machine_avr.go) but let's keep it as is for now - it can be changed when a different AVR comes along with a different PWM numbering system.